openml/study/functions.py

```diff
@@ -1,49 +1,84 @@
+from typing import cast, Dict, List, Optional, Union
+
+import dateutil.parser
 import xmltodict
 
-from openml.study import OpenMLStudy
+from openml.study import OpenMLStudy, OpenMLBenchmarkSuite
+from openml.study.study import BaseStudy
 import openml._api_calls
 
 
-def get_study(study_id, entity_type=None):
+def get_suite(suite_id: Union[int, str]) -> OpenMLBenchmarkSuite:
     """
-    Retrieves all relevant information of an OpenML study from the server
-    Note that some of the (data, tasks, flows, setups) fields can be empty
-    (depending on information on the server)
+    Retrieves all relevant information of an OpenML benchmarking suite from the server.
 
     Parameters
     ----------
     study id : int, str
         study id (numeric or alias)
 
     entity_type : str (optional)
-        Which entity type to return. Either {data, tasks, flows, setups,
-        runs}. Give None to return all entity types.
+        Do not change this!
+
+    Returns
+    -------
+    OpenMLSuite
+        The OpenML suite object
+    """
+    suite = cast(OpenMLBenchmarkSuite, _get_study(suite_id, entity_type='tasks'))
+    return suite
+
+
+def get_study(
+    study_id: Union[int, str],
+    arg_for_backwards_compat: Optional[str] = None,
+) -> OpenMLStudy:  # noqa F401
+    """
+    Retrieves all relevant information of an OpenML study from the server.
+
+    Parameters
+    ----------
+    study id : int, str
+        study id (numeric or alias)
 
     Returns
     -------
     OpenMLStudy
         The OpenML study object
     """
-    call_suffix = "study/%s" % str(study_id)
-    if entity_type is not None:
-        call_suffix += "/" + entity_type
+    if study_id == 'OpenML100':
+        study = _get_study(study_id, entity_type='tasks')
+        return cast(OpenMLBenchmarkSuite, study)  # type: ignore
+    else:
+        study = cast(OpenMLStudy, _get_study(study_id, entity_type='runs'))
+        return study
+
+
+def _get_study(id_: Union[int, str], entity_type) -> BaseStudy:
+    call_suffix = "study/{}".format(str(id_))
     xml_string = openml._api_calls._perform_api_call(call_suffix, 'get')
     force_list_tags = (
         'oml:data_id', 'oml:flow_id', 'oml:task_id', 'oml:setup_id',
         'oml:run_id',
         'oml:tag'  # legacy.
     )
-    result_dict = xmltodict.parse(xml_string,
-                                  force_list=force_list_tags)['oml:study']
+    result_dict = xmltodict.parse(xml_string, force_list=force_list_tags)['oml:study']
     study_id = int(result_dict['oml:id'])
     alias = result_dict['oml:alias'] if 'oml:alias' in result_dict else None
     main_entity_type = result_dict['oml:main_entity_type']
+    if entity_type != main_entity_type + 's':
+        raise ValueError(
+            "Unexpected entity type '{}' reported by the server, expected '{}'".format(
```

Comment:
probably also need to append the `'s'` in this message or it's going to get really confusing.